test/caller-test.jl

```diff
@@ -0,0 +1,46 @@
+function test_caller()
+    print("caller...")
+
+    R, (x, y) = PolynomialRing(Singular.ZZ, ["x", "y"])
+    @test Singular.SETS.isEqualInt(R, Singular.ZZ(1), Singular.ZZ(1)) == 1
+
+    R, (x, y) = PolynomialRing(Singular.QQ, ["x", "y"])
+
+    # Input tests
+    @test Singular.SETS.isEqualInt(R, 1, 1)                           == 1
+    @test Singular.SETS.isEqualInt(R, x, x)                           == 1
+    @test Singular.SETS.isEqualInt(R, "aa", "aa")                     == 1
+    @test Singular.SETS.isEqualInt(R, Singular.QQ(1), Singular.QQ(1)) == 1
+    @test Singular.SETS.isEqualInt(R, [1,2,3], [1,2,3])               == 1
+    @test Singular.SETS.isEqualInt(R, [1 2; 3 4], [1 2; 3 4])         == 1
+
+    R, (x,y,z) = PolynomialRing(Singular.QQ, ["x", "y", "z"])
+    
+    i1 = Singular.POLY.cyclic(R, 3)
+    i2 = Ideal( R, x+y+z, x*y+x*z+y*z, x*y*z-1 )
+    @test equal(i1, i2)
+
+    vec = FreeModule(R,2)([x,y])
+    mod = Singular.Module(R, vec)
+    i1 = Singular.POLY.mod2id(R,mod,[1,2])
```

Comment:
I'm not particularly fond of all CAPS names. It is not done in Julia in general.

Reply:
The main problem is that I could not add for example a `Poly` submodule to the `Singular` module, because Module names need to be unique and cannot be types or whatever. So whichever other (reasonable) spelling I tried (all small, first letter capitalized) it always failed. This is why I did all-caps. The only solution would be to put the functions in a different module, say `Libs`. But then the names get longer. If you have a different suggestion, it is highly welcome.

Reply:
Ping on the naming stuff here. Did anybody come up with a better naming convention?